docs/interactive_engine/tutorial_ldbc_gremlin.md

```diff
@@ -831,3 +831,1087 @@ At last, we would like to introduce `endV()` step in path expansion. Since a pat
 
 Figure 18. The examples of path expansion.
 :::
+### Real Applications
+
+Graph traversal enables us to conduct more complicated data analytics on the graph. 
+
+For example, the path `comment-hasCreator->person->isLocatedIn->city-isPartOf->country` can help to determine the nationality of `comments`. Therefore,  we can calculate the histogram of comment length for comments from China easily with GIE:
+
+```python
+import matplotlib.pyplot as plt
+import numpy as np
+import graphscope as gs
+from graphscope.dataset.ldbc import load_ldbc
+
+# load the modern graph as example.
+graph = load_ldbc()
+
+# Hereafter, you can use the `graph` object to create an `gremlin` query session
+g = gs.gremlin(graph)
+# Extract all comments' contents in China
+q1 = g.execute('g.V().hasLabel(\'person\').where(out(\'isLocatedIn\').out(\'isPartOf\').values(\'name\').is(eq(\'China\'))).in(\'hasCreator\').values(\'content\')')
+comment_contents = q1.all()
+comment_length = [len(comment_content.split()) for comment_content in comment_contents];
+
+plt.hist(x=list(filter(lambda x:(x< 50), comment_length)), bins='auto', color='#607c8e',alpha=0.75)
+plt.grid(axis='y', alpha=0.75)
+plt.xlabel('Comments Length')
+plt.ylabel('Comments Count')
+plt.title('Comment Length Histogram for comments from China')
+plt.show()
+```
+:::{figure-md}
+![comments_length_histogram_china](../images/comments_length_histogram_china.png)
+Figure 19. Comments length histogram in China.
+:::
+
+Or we can estimate the browser market share in Japan:
+
+```python
+import matplotlib.pyplot as plt
+import graphscope as gs
+from graphscope.dataset.ldbc import load_ldbc
+from collections import Counter
+
+
+# load the modern graph as example.
+graph = load_ldbc()
+
+# Hereafter, you can use the `graph` object to create an `gremlin` query session
+g = gs.gremlin(graph)
+
+# Extract all person' gender property value
+q1 = g.execute('g.V().hasLabel(\'person\').where(out(\'isLocatedIn\').out(\'isPartOf\').values(\'name\').is(eq(\'Japan\'))).in(\'hasCreator\').values(\'browserUsed\')')
+browsers_used = q1.all()
+browser_list = ['Firefox', 'Chrome', 'Internet Explorer', 'Safari']
+# Count Firefox, Chrome, Internet Explorer and Safari
+browser_counts = Counter(browsers_used)
+
+# Draw pie chart
+plt.pie(x=[browser_counts[browser] for browser in browser_list], labels=browser_list, autopct='%1.1f%%')
+plt.show()
+```
+:::{figure-md}
+![estimated_japan_brower_market_share](../images/estimated_japan_brower_market_share.png)
+Figure 20. Estimated Browser Market Share in Japan.
+:::
+
+## Pattern Match
+
+Suppose you want to find two `persons` and the `universities` they `studyAt` in the Property Graph that:
+
+1. The two `person` `know` each other
+2. The two `person` `studyAt` the same `university`
+
+With the previously taught knowledge, you may write the following code in GIE to achieve this purpose:
```

Comment:
Recommend proof-reading using chatgpt-like tools. The words can be improved a bit.